### PR TITLE
KEP-1577 Fix issue where startup timeout always fired

### PR DESCRIPTION
**Pull request by nickroeker**

### CODE REVIEW

File: src/daemonManager.js [hunk 1]
```diff
@@ -145,19 +145,26 @@ const generateDaemon = (swarmId, daemonConfig) => {
             }
         });
 
-        await new Promise((resolve, reject) => {
-            setTimeout(() => {
-                getDaemonProcess().kill();
-                reject(new Error(`Daemon-${daemonConfig.listener_port} failed to start in ${harnessConfigs.daemonStartTimeout}ms.`))
-            }, harnessConfigs.daemonStartTimeout);
-
+        let startupTimeout = new Promise((_, reject) => {
```
Comment (jyclam) [inline src/daemonManager.js]: Should you use const instead of let since you never reassign the variable?

File: src/daemonManager.js [hunk 2]
```diff
@@ -145,19 +145,26 @@ const generateDaemon = (swarmId, daemonConfig) => {
...
+            setTimeout(
+                () => reject(new Error(`Daemon-${daemonConfig.listener_port} failed to start in ${harnessConfigs.daemonStartTimeout}ms.`)),
+                harnessConfigs.daemonStartTimeout
+            );
+        });
+        let startupSuccess = new Promise((resolve) => {
```
Comment (jyclam) [inline src/daemonManager.js]: Should you use const instead of let since you never reassign the variable?

File: src/daemonManager.js (hunk 1 above)
Comment (nickroeker) [reply src/daemonManager.js]: Yepp I should, incoming patch

File: src/daemonManager.js (hunk 2 above)
Comment (nickroeker) [reply src/daemonManager.js]: Yepp I should, incoming patch

**Review by jyclam — APPROVED** [review]

**Pull request merged**